test/sqllogictest/list.slt

```diff
@@ -2575,3 +2575,242 @@ SELECT LIST[LIST[customer.first_name, customer.last_name], LIST[customer.zip], L
 ----
 {{alice,lasta},{10003},{1}}
 {{charlie,lastc},{11217},{3}}
+
+## Optimize list_index(list_create, literal) and multidimensional variations
```

Comment:
Maybe add a comment explaining why these tests are here and not in `src/expr/tests/testdata`

Reply:
done